ckan/controllers/package.py

```diff
@@ -667,8 +667,8 @@ def new_resource(self, id, data=None, errors=None, error_summary=None):
                 except NotFound:
                     abort(404, _('The dataset {id} could not be found.'
                                  ).format(id=id))
-                if not len(data_dict['resources']):
-                    # no data so keep on page
+                if asbool(config.get('ckan.dataset.create.require.resource', 'True')) and not len(data_dict['resources']):
```

Comment:
Can you rename the option to something like `ckan.dataset.create_on_ui_requires_resources`?

Reply:
This line should be shorter or the PEP8 test will complain.

Reply:
I have made those two changes

Reply:
@howff these are awesome changes, I'm looking forward to them in the next patch 👍 

I noticed the pep8 tests are still failing after your changes to `package.py` at lines 670, 671 and 672. 

```
> ---------------------------
> ckan/controllers/package.py
> ---------------------------
> E501 ln:670 line too long (95 > 79 characters)
> E125 ln:671 continuation line with same indent as next logical line
> E501 ln:672 line too long (80 > 79 characters)
> 
> Failing Files:
> ==============
> ckan/controllers/package.py
```

This should (I hope) fix it (I did a quick test of the whole file before and after at http://pep8online.com/, there's still some warnings but no more failures):

```
                if asbool(config.get(
                        'ckan.dataset.create_on_ui_requires_resources',
                        'True')) \
                        and not len(data_dict['resources']):
                    # no data and configured to require resource so keep on 
                    # page
```

Reply:
Can't you turn that error E501 into a warning? I am one of those people who edits code in a terminal window so this error message should be to my benefit. However it's actually creating code which is less readable. And at some point it won't even be possible to fit into 79 chars (a long config option with several levels of indenting) so what happens then? I am also surprised that a one-line addition is creating so much round-tripping and extra work for so many people; I naively imagined that making a contribution would be much easier than this ;-)